### ISSUE TITLE
📢 [提交前请务必阅读] Read this before submitting issue 

### ISSUE THREAD

**Issue by weicj** [weicj/vLLM-2080Ti-Definitive#74]
**语言规范：根据Github社区要求，为保障全球开发者的检索效率，本仓库相关问题请尽量以英文反馈，如有不便敬请见谅。**

**问题边界：本项目聚焦于核心软件与推理优化，有关硬件选购、基础安装等外延讨论不在本仓库受理范围内。**

**完整反馈：反馈问题时请务必提供完整的系统背景、硬件拓扑与错误日志，并清晰表达问题的产生过程与复现路径。**

**最佳实践：欢迎分享最佳实践结果，并附上系统环境、实测性能等数据，作为公众参考环境反馈。**

**保持尊重：本项目为开源项目，仅用于个人学习之用，不保证仓库所宣称的各项性能指标能在不同硬件环境下稳定复现。**

**抓住共性：在本人力所能及的范围内，会尽量就共性问题提供解决思路，由于个人精力所限，不提供一对一的技术服务。**

**规范执行：请参考下文回复的模板进行完整情况说明，若您面临语言或表达困扰，请借助AI工具将文本润色完善后再行提交。**

**日常基础问题请进入群组讨论 General Discussion add Discord Group : https://discord.gg/VFqVVySdMS**

---
To ensure efficient project maintenance and to foster a globally accessible community, the maintainers require all users to adhere to the following guidelines when opening a new issue. 

Managing an open-source project involves balancing active development with community support. By providing clear, structured technical data, you help us triage bugs faster and keep the project moving forward.

---

### 1. 🌐 Language Policy
To align with GitHub's global practices and ensure discussions are searchable for the entire community, **the maintainer recommend you submit issues in English.**

* All text (descriptions, reproduction steps, expected behavior) are preferrably written in English.
* *Note: Raw system logs, error traces, and hardware configurations can remain in their original format.*

### 2. 🛑 Project Scope & Target Audience
This repository is tailored for users who already possess foundational knowledge in hardware engineering and system administration. **Discussions regarding basic purchasing, installation, or generic hardware assembly are strictly out of scope.** Before participating, you are expected to independently handle:
* **Interconnection & Topology:** Configuring peer-to-peer (P2P) communication, NVLink/PCIe topologies, and multi-GPU clustering.
* **Infrastructure Basics:** Resolving power supply requirements (EPS/PCIe budgeting) and thermal/cooling management (e.g., shroud designs, airflow optimization).
* **System Dependencies:** Troubleshooting low-level driver stacks (e.g., CUDA, ROCm, kernel modules) and basic OS environments.

*Issues asking for hardware procurement recommendations, shopping advice, or step-by-step rigging guides will be closed immediately.*

### 3. 🛠️ Required Technical Context
We cannot diagnose software or inference anomalies without an established baseline. Due to the high dependency on individual hardware setups, **this repository does not guarantee performance reproducibility across different environments.** You must provide:
* **Complete System Background:** Detail your specific processor/hardware models, driver versions, and framework stack.
* **Topology & Infrastructure:** Clearly describe your hardware topology (e.g., interconnection, device relationships) or environment setup.
* **Full Error Logs:** Paste the comprehensive, unedited runtime logs or stack traces. Please use markdown code blocks instead of posting screenshots of text.

### 4. 🚦 Policy Enforcement & AI Assistance
To keep our backlog manageable and maintain repository health, issues that fail to meet these criteria—such as those missing logs, lacking environment descriptions, focusing on out-of-scope hardware procurement, or written in non-English languages—**may be closed automatically by the maintainers or automated bots.**

* **AI & Agent Assistance:** If you are unsure how to clearly articulate, translate, or structure your technical problem to meet these guidelines, we highly encourage utilizing AI assistants or agents (e.g., ChatGPT, Claude) to refine and draft your text prior to submission. 

---

**Thank you for respecting the maintainers' time and for your cooperation in making this project more robust and accessible for everyone!**

**Comment by weicj** [comment]
# Issue Submission Template

This document is a reference format for new issues in this repository. It is intended to help reporters provide the minimum technical context required by maintainer. It is highly suggested you drop the following template into AI agent (Codex/Claude Code/OpenCode..) to help you draft the body in every case.

## Rules

- Paste logs as plain text inside fenced code blocks. Do not upload screenshots of text logs unless the failure only exists as an image.
- Include the full system background, hardware topology, exact reproduction path, and the relevant error output.
- Keep out-of-scope topics out of this tracker: hardware shopping advice, generic rig assembly, and basic system administration questions should not be filed here.

## Copy-Paste Template

````md
## Issue Type
- build failure / runtime failure / quality regression / performance regression / best-practice report

## Summary
Describe the problem in 2-5 sentences.

## Reproduction Steps
1. Start from a clean or existing environment: ...
2. Run: ...
3. Select profile / mode / model: ...
4. Observe the failure: ...

## Expected Behavior
What should have happened?

## Actual Behavior
What actually happened?

## Exact Command / Profile / Model
- launcher mode:
- profile path:
- model path or model name:
- exact command:
- whether this is a fresh build or an existing runtime:

## System Environment
- OS:
- kernel:
- CPU:
- RAM:
- GPU(s):
- target GPU pair used for this run:
- GPU topology / NVLink / P2P:
- NVIDIA driver:
- CUDA toolkit (`nvcc --version`):
- Python:
- PyTorch:
- fork release / commit:

## Logs
```text
Paste the full relevant log here.
```

## Additional Notes
Any extra observations, workarounds tried, or differences from the validated setup.
````

## Recommended Extra Data by Issue Type

### Build Failure

Also include:

- whether you used `build.sh` directly
- whether the failure happens on a fresh clone
- the last 200 lines of the latest build log

```bash
tail -n 200 "$(ls -t build-logs/build-*.log | head -1)"
```

### Runtime Failure

Also include:

- whether the server was started by `launcher.sh` or manually
- the selected mode, profile, KV route, and model
- whether the problem happens on first request or only after warmup / graph capture
- the exact request shape if relevant, such as prompt length, output length, image count, MTP on/off, tool calling on/off

### Performance Regression

Also include:

- test shape, for example `4096/128` or `64K prompt + 512 decode`
- prefill tok/s
- decode tok/s
- whether the run is noMTP or MTP
- whether the route is `safe`, `normal`, `fast`, or `aggressive`
- whether this is a fresh graph / cold cache run or a warmed run

Use a compact format like:

```text
profile: qwen27b/normal/fp8/fp16kv-256K-mtp3-text-only.env
shape: 4096 / 128
mode: normal
mtp: MTP3
prefill: 1580 tok/s
decode: 96 tok/s
notes: fresh build, first warmed run
```

### Best-Practice Report

If you are sharing a successful setup, include:

- system environment
- exact GPUs used
- model and profile
- maximum validated context
- measured prefill and decode throughput
- any special constraints, such as power limit, display attached, or no NVLink

## Helpful Collection Commands

### System Baseline

```bash
uname -a
cat /etc/os-release
python3 --version
nvidia-smi
nvidia-smi topo -m
nvcc --version
```

### Python / Torch Baseline

```bash
python3 - <<'PY'
import sys
try:
    import torch
    print("python:", sys.version)
    print("torch:", torch.__version__)
    print("torch.cuda:", torch.version.cuda)
    print("cuda available:", torch.cuda.is_available())
    print("device count:", torch.cuda.device_count())
except Exception as exc:
    print("torch probe failed:", exc)
PY
```

### Build Log Tail

```bash
tail -n 200 "$(ls -t build-logs/build-*.log | head -1)"
```

## Notes for Reporters

- One issue should describe one main problem. Do not combine unrelated build, runtime, and benchmark topics in the same report.
- If your environment differs substantially from the validated reference lane, say so explicitly instead of assuming the maintainer can infer it from the logs.

**Comment by weicj** [comment]
# Issue 提交参考模板

这是一份给中文用户参考的 issue 提交模板说明文档，用来帮助整理信息并满足问题反馈的要求。

请注意：**本仓库依然优先建议您使用英文提交 issue。** 如您在语言方面遇到难以解决的问题，依然可以提交中文反馈。这份中文模板的核心目的是帮助你把问题、环境、复现路径、日志说清楚，在后续问题反馈中，建议尽量保持这个结构。

## 基本规则

- 问题摘要、复现步骤、预期行为、实际行为这些核心部分需要表达清晰。
- 日志请使用 Markdown 代码块直接粘贴文本，不要用日志截图代替文本日志。
- 请提供完整的系统背景、硬件拓扑、复现路径，以及相关错误输出。
- 硬件采购建议、装机指导、基础系统运维问题不属于本仓库 issue 处理范围。
- 优先使用AI Agent工具帮你梳理问题草案（直接把这个模板丢给他让他帮你写，我们既然想用AI，就要习惯用AI去帮助我们提升效率）。

## 可直接套用的结构模板

````md
## 问题类型
- 构建失败 / 运行时故障 / 输出质量异常 / 性能回退 / 最佳实践分享

## 问题概要
请用 2-5 句话说明你遇到了什么问题。

## 复现步骤
1. 当前环境是全新环境还是已有环境：...
2. 执行了什么命令：...
3. 选择了什么 profile / mode / 模型：...
4. 在哪一步出现问题：...

## 预期行为
你原本期望发生什么？

## 实际行为
实际发生了什么？

## 使用的命令 / Profile / 模型
- launcher 模式：
- profile 路径：
- 模型路径或模型名称：
- 完整命令：
- 这是全新 build 还是已有 runtime：

## 系统环境
- 操作系统：
- 内核：
- CPU：
- 内存：
- GPU：
- 本次实际使用的目标 GPU 对：
- GPU 拓扑 / NVLink / P2P：
- NVIDIA 驱动版本：
- CUDA toolkit（`nvcc --version`）：
- Python：
- PyTorch：
- fork 版本 / commit：

## 日志
```text
Paste the full relevant log here.
```

## 补充说明
这里可以写额外观察、已经尝试过的排查动作、和 README 已验证环境相比有哪些差异。
````

## 各类问题建议补充的信息

### 1. 编译错误

除了上面的基本模板，建议额外补充：

- 是否直接使用 `build.sh`
- 是否在全新 clone / 全新环境下复现
- 最新 build 日志最后 200 行

```bash
tail -n 200 "$(ls -t build-logs/build-*.log | head -1)"
```

### 2. 运行时错误

建议额外补充：

- 服务是通过 `launcher.sh` 启动还是手动启动
- 使用的 mode、profile、KV 路线、模型名称
- 问题是在首个请求就发生，还是 warmup / graph capture 之后才出现
- 如果和请求形状有关，请说明 prompt 长度、输出长度、图片数量、MTP 开关、tool calling 开关等

### 3. 性能回退

建议额外补充：

- 测试口径，例如 `4096/128` 或 `64K prompt + 512 decode`（如连续生成’the'，排除命中率干扰）
- prefill tok/s
- decode tok/s
- 是否为 noMTP 或 MTP
- 使用的是 `safe`、`normal`、`fast` 还是 `aggressive`
- 是冷启动 / 冷图缓存，还是预热后的结果

建议用这种紧凑格式：

```text
profile: qwen27b/normal/fp8/fp16kv-256K-mtp3-text-only.env
shape: 4096 / 128
mode: normal
mtp: MTP3
prefill: 1580 tok/s
decode: 96 tok/s
notes: fresh build, first warmed run
```

### 4. 最佳实践分享

如果你是在分享成功经验，建议补充：

- 系统环境
- 实际使用的 GPU
- 模型与 profile
- 最大已验证上下文
- 实测 prefill 与 decode 吞吐
- 特殊约束，例如限功耗、接显示器、无 NVLink 等

## 推荐采集命令

### 系统基础信息

```bash
uname -a
cat /etc/os-release
python3 --version
nvidia-smi
nvidia-smi topo -m
nvcc --version
```

### Python / Torch 基础信息

```bash
python3 - <<'PY'
import sys
try:
    import torch
    print("python:", sys.version)
    print("torch:", torch.__version__)
    print("torch.cuda:", torch.version.cuda)
    print("cuda available:", torch.cuda.is_available())
    print("device count:", torch.cuda.device_count())
except Exception as exc:
    print("torch probe failed:", exc)
PY
```

### Build 日志尾部

```bash
tail -n 200 "$(ls -t build-logs/build-*.log | head -1)"
```

## 提交前建议

- 一个 issue 只讲一个主要问题，不要把 build、runtime、benchmark 等多个无关问题混在一起。
- 如果你的环境和仓库 README 中的已验证参考环境差异很大，请主动写明，不要默认维护者能从零散日志里自己推断出来，感谢理解
